File: python code/binary image processing/opening.py
import cv2

# read image
image = cv2.imread("../../images/opening.png", cv2.IMREAD_GRAYSCALE)

# initialize number of iteration
iterations = 3

# get structuring kernel
openingSize = 3
element = cv2.getStructuringElement(shape=cv2.MORPH_ELLIPSE,
                                    ksize=(2 * openingSize + 1, 2 * openingSize + 1),
                                    anchor=(openingSize, openingSize))

# apply morphological opening
morphOpenedImage = cv2.morphologyEx(src=image,
                                    op=cv2.MORPH_OPEN,
                                    kernel=element,
                                    iterations=3)

# display images
cv2.imshow("Original Image", image)
cv2.imshow("Opening Image", morphOpenedImage)

cv2.waitKey(0)

File: c++ code/week1/examples/binary image processing/binary image processing/opening.cpp
#include <opencv2/opencv.hpp>

int main4()
{
	// read image
	cv::Mat image = cv::imread("../../../../../images/opening.png", cv::IMREAD_GRAYSCALE);

	// initialize number of iterations
	int iterations = 3;

	// create struturing element
	int openingSize = 3;
	cv::Mat element = cv::getStructuringElement(cv::MORPH_ELLIPSE,
		cv::Size(2 * openingSize + 1, 2 * openingSize + 1),
		cv::Point(openingSize, openingSize));

	// apply morphological opening
	cv::Mat morphOpenedImage;
	cv::morphologyEx(image, morphOpenedImage, cv::MORPH_OPEN, element, cv::Point(-1, -1), iterations);

	// display images
	cv::imshow("Original Image", image);
	cv::imshow("Opening Image", morphOpenedImage);

	cv::waitKey(0);

	return EXIT_SUCCESS;
}